### PR TITLE
suppress unreachable string format errors

### PR DESCRIPTION
**Pull request by etan-status**
We use `%` and `&` to format some strings, which may raise `ValueError` if the format string is weird. As we are using them in ways that should always succeed, catch those `ValueError` and convert to asserts.

### CODE REVIEW

File: confutils.nim [hunk 1]
```diff
@@ -394,8 +394,14 @@ func getNextArgIdx(cmd: CmdInfo, consumedArgIdx: int): int =
   -1
 
 proc noMoreArgsError(cmd: CmdInfo): string =
-  result = if cmd.isSubCommand: "The command '$1'" % [cmd.name]
-           else: appInvocation()
+  result =
+    if cmd.isSubCommand:
+      try:
+        "The command '$1'" % [cmd.name]
+      except ValueError as err:
```
Comment (tersec) [inline confutils.nim:401]: With this exception caught, are any other exceptions possible from `noMoreArgsError`? If not, then its signature can be annotated not to throw exceptions.

File: confutils.nim [hunk 2]
```diff
@@ -856,8 +862,10 @@ proc addConfigFileContent*(secondarySources: auto,
 func constructEnvKey*(prefix: string, key: string): string =
   ## Generates env. variable names from keys and prefix following the
   ## IEEE Open Group env. variable spec: https://pubs.opengroup.org/onlinepubs/000095399/basedefs/xbd_chap08.html
-
-  return (&"{prefix}_{key}").toUpperAscii.multiReplace(("-", "_"), (" ", "_"))
+  try:
+    (&"{prefix}_{key}").toUpperAscii.multiReplace(("-", "_"), (" ", "_"))
+  except ValueError as err:
```
Comment (tersec) [inline confutils.nim:867]: Same question here, about `constructEnvKey`

File: confutils.nim (hunk 1 above)
Comment (etan-status) [reply confutils.nim:401]: 1c4679b18e71875bdcd51d96d77442701e4df6fe

File: confutils.nim (hunk 2 above)
Comment (etan-status) [reply confutils.nim:867]: 1c4679b18e71875bdcd51d96d77442701e4df6fe

**Review by tersec — APPROVED** [review]

**Comment by arnetheduck** [comment]
not necessarily for this PR, but the module lacks {.push raises... I think

**Comment by etan-status** [comment]
Yeah, not possible yet, needs some of the other PRs before we can do that. For now, goal is to unblock https://github.com/status-im/nimbus-eth2/pull/5010

**Pull request merged**